lib/Models/Catalog/CatalogItems/GeoJsonCatalogItem.ts

```diff
@@ -74,19 +75,42 @@ class GeoJsonCatalogItem extends GeoJsonMixin(
           this.requestData ? toJS(this.requestData) : undefined,
           this.postRequestDataAsFormData
         );
+        if (this.responseDataPath) {
+          jsonData = _get(jsonData, this.responseDataPath);
+        }
       }
     }
 
+    if (jsonData === undefined) {
+      throw TerriaError.from("Failed to load geojson");
+    }
+
+    let fc;
     // Transform jsonData to feature collection
-    if (isJsonObject(jsonData) && typeof jsonData.type === "string") {
-      const fc = toFeatureCollection(jsonData);
-      if (fc) return fc;
-      throw TerriaError.from(
-        "Invalid geojson data - only FeatureCollection and Feature are supported"
+    if (Array.isArray(jsonData)) {
+      // Array that isn't a feature collection
+      fc = toFeatureCollection(
+        jsonData.map(item => {
+          let geojson = this.responseGeoJsonPath
+            ? _get(item, this.responseGeoJsonPath)
+            : item;
+          if (typeof geojson === "string") {
+            geojson = JSON.parse(geojson);
+          }
+          // add extra properties back to geojson so they appear in feature info
+          geojson.properties = item;
```

Comment:
Should this exclude the object at responseGeoJsonPath and include just the outer item?

Reply:
ah yeah you're totally right let's do that